### PR TITLE
Add uninstall script

### PR DESCRIPTION
**Pull request by rany2**

### CODE REVIEW

File: uninstall.sh [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+
+# Installation script for cake-autorate.sh
```
Comment (lynxthecat) [inline uninstall.sh:3]:
```suggestion
# Uninstall script for cake-autorate.sh
```

Comment (lynxthecat) [reply uninstall.sh]: Could we make the same change for setup.sh, i.e. "Installation script for cake-autorate"?

**Comment by lynxthecat** [comment]
What's the thinking underlying separating this out from setup.sh?

**Comment by rany2** [comment]
> What's the thinking underlying separating this out from setup.sh?

Seems easier to maintain.

File: uninstall.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	then
+		/etc/init.d/cake-autorate stop || true
+	fi
+	rm -f /etc/init.d/cake-autorate /etc/rc.d/*cake-autorate
+
+	# Check if an instance of cake-autorate is already running and exit if so
+	if [ -d /var/run/cake-autorate ]
+	then
+		printf "cake-autorate is already running - exiting\n" >&2
+		printf "If you want to install a new version, stop the service first\n" >&2
```
Comment (lynxthecat) [inline uninstall.sh:27]:
```suggestion
		printf "If you want to install a new version, first stop any running instance of cake-autorate\n" >&2
```

File: uninstall.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	if [ -x /etc/init.d/cake-autorate ]
+	then
+		/etc/init.d/cake-autorate stop || true
+	fi
+	rm -f /etc/init.d/cake-autorate /etc/rc.d/*cake-autorate
+
+	# Check if an instance of cake-autorate is already running and exit if so
+	if [ -d /var/run/cake-autorate ]
+	then
+		printf "cake-autorate is already running - exiting\n" >&2
```
Comment (lynxthecat) [inline uninstall.sh:26]:
```suggestion
		printf "An instance of cake-autorate appears to be already running - exiting\n" >&2
```

Comment (rany2) [reply uninstall.sh:26]: Technically, it could be multiple instances

File: uninstall.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		printf "If you want to install a new version, stop the service first\n" >&2
+		printf "If you are sure cake-autorate is not running, delete the /var/run/cake-autorate directory\n" >&2
```
Comment (lynxthecat) [inline uninstall.sh:28]:
```suggestion
		printf "If you are sure that no instance of cake-autorate is running, delete the /var/run/cake-autorate directory\n" >&2
```

File: uninstall.sh (hunk 2 above)
Comment (rany2) [reply uninstall.sh:27]: Well, its an uninstaller. That disclaimer needs to be changed.

File: uninstall.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	if [ -x /etc/init.d/cake-autorate ]
+	then
+		/etc/init.d/cake-autorate stop || true
+	fi
+	rm -f /etc/init.d/cake-autorate /etc/rc.d/*cake-autorate
+
+	# Check if an instance of cake-autorate is already running and exit if so
+	if [ -d /var/run/cake-autorate ]
+	then
+		printf "An instance of cake-autorate appears to be already running - exiting\n" >&2
```
Comment (lynxthecat) [inline uninstall.sh:26]:
```suggestion
		printf "At least one instance of cake-autorate appears to be running - exiting\n" >&2
```

**Comment by lynxthecat** [comment]
Shall I still merge notwithstanding the workflow issue?

**Comment by rany2** [comment]
Sure

**Pull request merged**